### PR TITLE
I3305 surface window sizes in api

### PR DESCRIPTION
**Pull request by mlholland**
# Description

These changes add a new plugin which surfaces two new values in the gloo api. These values are initial_stream_window_size and initial_connection_window_size from [envoy's http2protocoloptions](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/protocol.proto#envoy-v3-api-msg-config-core-v3-http2protocoloptions).

Questions for reviewers:
 - should this new plugin contain or use any other values, like useHttp2?
 - the grpc plugin has a test to ensure that no other upstream assigns the http2protocoloptions struct to exist. Should that be removed?

# Context

https://github.com/solo-io/gloo/issues/3305

# Checklist:

- [X] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [X] If I updated APIs (our protos) or helm values, I ran `make update-deps generated-code` to ensure there will be no code diff
- [X] I followed guidelines laid out in the Gloo [contribution guide](https://docs.solo.io/gloo/latest/contributing/
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/3305

### CODE REVIEW

**Comment by mlholland** [comment]
/kick - most recent error looks like cleanup issue from another run

**Comment by kdorosh** [comment]
> should this new plugin contain or use any other values, like useHttp2?

i believe we already handle this on the upstream and special case this config, rather than add it to it's own plugin

File: changelog/v1.5.0-beta7/surface-window-size-http-options.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+changelog:
+    - type: NEW_FEATURE
+      issueLink: https://github.com/solo-io/gloo/issues/3305
+      description: Expose InitialStreamWindowSize and InitialConnectionWindowSize as http2 protocol options. They both default to 268435456.
```
Comment (kdorosh) [inline changelog/v1.5.0-beta7/surface-window-size-http-options.yaml:4]: 268435456 what? let's make this more user friendly

Comment (kdorosh) [reply changelog/v1.5.0-beta7/surface-window-size-http-options.yaml:4]: 256 MiB

Comment (mlholland) [reply changelog/v1.5.0-beta7/surface-window-size-http-options.yaml]: corrected

**Pull request merged**